src/app/projects/project-list/project-list.component.ts

```diff
@@ -10,27 +12,99 @@ import { ProjectService } from '../project.service';
 export class ProjectListComponent implements OnInit {
   errors: any[] = [];
   projectList: any[] = [];
-  donatedAmount = {};
-  color = 'primary';
-  mode = 'determinate';
+  project: Project;
+  currentUser: any;
+
+  isRed = false;
+  isGray = true;
 
   donateOption1 = 10;
   donateOption2 = 25;
   donateOption3 = 50;
-  constructor(private _projectService: ProjectService) { }
+
+  constructor(private _projectService: ProjectService,
+              private _localStorageService: LocalStorageService) {
+  }
 
   ngOnInit() {
 
+    this.currentUser = this._localStorageService.getCurrentUser();
+
     this._projectService.getProjects().subscribe(
       res => {
         console.log(res);
         this.projectList = res.filter((v, k) => {
-          return v.open === 'true';
-
-        },
+            return v.open === 'true';
+          },
           error => this.errors.push(error)
         );
       }
-    )
+    );
+  }
+
+
+  /**
+   * this function will return a class that is responsible for activating and deactivating the color red in the
+   * like button
+   * @param {number} project_id
+   * @returns {any}
+   */
+  likeButtonColor(project_id: number): any {
+    const classes = {
+      'heart-red': this.isRed,
+      'heart-gray': this.isGray,
+    };
+
+    if (this.currentUser !== null || this.currentUser !== undefined) {
+      if (this.currentUser.savedProject == null || this.currentUser.savedProject === undefined) {
+        classes['heart-red'] = false;
+        classes['heart-gray'] = true;
+        return classes;
+
+      } else {
+        if (this.currentUser.savedProject.indexOf(project_id) > -1) {
+          classes['heart-red'] = true;
```

Comment:
#182 RTBM but you can create function instead of repeat  many times.
 classes['heart-red'] = false;
 classes['heart-gray'] = true;